foxpuppet/windows/browser/bookmarks/bookmark.py

```diff
@@ -0,0 +1,213 @@
+        """Create a bookmark object.
+
+        Args:
+            window (:py:class:`BrowserWindow`): Window object this bookmark appears in
+            root (:py:class:`~selenium.webdriver.remote.webelement.WebElement`): WebDriver element object for bookmark
+
+        Returns:
+            :py:class:`Bookmark`: Bookmark instance
+        """
+        with window.selenium.context(window.selenium.CONTEXT_CHROME):
+            return Bookmark(window, root)
+
+    @property
+    def is_bookmarked(self) -> bool:
+        """Checks if the current page is bookmarked using the star button.
+
+        Returns:
+            bool: True if the page is bookmarked, False otherwise.
+        """
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            star_button_image = self.selenium.find_element(
+                *BookmarkLocators.STAR_BUTTON_IMAGE
+            )
+            return star_button_image.get_attribute("starred") == "true"
+
+    def add_bookmark(
+        self, bookmark_data: Optional[BookmarkData] = None, is_detailed: bool = False
+    ) -> None:
+        """
+        Add a bookmark using either quick add (star button) or detailed menu approach.
+
+        Args:
+            detailed (bool, optional): Whether to use detailed menu approach. Defaults to False.
+            bookmark_data (BookmarkData, optional): Data for the bookmark when using detailed menu.
+                Required when detailed is True.
+        """
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            if not is_detailed:
+                self.selenium.find_element(*BookmarkLocators.STAR_BUTTON).click()
+                self.selenium.find_element(*BookmarkLocators.FOLDER_MENU).click()
+                self.selenium.find_element(*BookmarkLocators.OTHER_BOOKMARKS_STAR).click()
+                self.selenium.find_element(*BookmarkLocators.SAVE_BUTTON).click()
+            else:
+                with self.selenium.context(self.selenium.CONTEXT_CHROME):
+                    self.actions.context_click(
+                        self.selenium.find_element(*BookmarkLocators.NAVIGATOR_TOOLBOX)
+                    ).perform()
+                    self.selenium.find_element(*BookmarkLocators.MENU_BAR).click()
+                    self.selenium.find_element(
+                        *BookmarkLocators.MAIN_MENU_BOOKMARK
+                    ).click()
+                    self.actions.context_click(
+                        self.selenium.find_element(*BookmarkLocators.MANAGE_BOOKMARKS)
+                    ).perform()
+                    self.selenium.find_element(*BookmarkLocators.ADD_BOOKMARK).click()
+
+                    frame_element = self.selenium.find_element(
+                        *BookmarkLocators.ADD_BOOKMARK_FRAME
+                    )
+                    self.selenium.switch_to.frame(frame_element)
+                    if bookmark_data is not None:
+                        if bookmark_data["name"]:
```

Comment:
I think you could use the walrus operator here, and it would allow `bookmark_data` to be used bellow as well.